src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -446,37 +458,34 @@ public boolean configure(final StaplerRequest req, final JSONObject formData) th
                 this.setTargetLogCollectionPort(null);
             }
 
-            try {
-                final String traceCollectionPortStr = formData.getString("targetTraceCollectionPort");
-                if(validatePort(traceCollectionPortStr)){
-                    this.setTargetTraceCollectionPort(formData.getInt("targetTraceCollectionPort"));
-                }else{
-                    this.setTargetTraceCollectionPort(null);
-                }
-            } catch (Exception e) {
-                // As there is no public UI to configure this property,
-                // the value is set to the default port to trace collection.
-                // formData.getString throws an exception
-                // if the key to search does not exist.
-                // NOTE: Change this when APM Traces was released as public feature.
+            final String traceCollectionPortStr = formData.getString("targetTraceCollectionPort");
+            if(validatePort(traceCollectionPortStr)){
+                this.setTargetTraceCollectionPort(formData.getInt("targetTraceCollectionPort"));
+            }else{
                 this.setTargetTraceCollectionPort(DEFAULT_TRACES_PORT_VALUE);
             }
 
             try {
-                final String traceServiceName = formData.getString("traceServiceName");
-                if(StringUtils.isNotBlank(traceServiceName)){
-                    this.setTraceServiceName(traceServiceName);
-                } else {
-                    this.setTraceServiceName(DEFAULT_TRACES_SERVICE_NAME);
+                final JSONObject ciVisibilityData = formData.getJSONObject("ciVisibilityData");
+                if(ciVisibilityData != null && !ciVisibilityData.isNullObject()) {
+                    if(!"DSD".equalsIgnoreCase(reportWith)) {
+                        throw new FormException("CI Visibility can only be enabled using Datadog Agent mode.", "collectBuildTraces");
+                    }
+
+                    final String ciInstanceName = ciVisibilityData.getString("traceServiceName");
+                    if(StringUtils.isNotBlank(ciInstanceName)){
+                        this.setCiInstanceName(ciInstanceName);
+                    } else {
+                        this.setCiInstanceName(DEFAULT_CI_INSTANCE_NAME);
+                    }
                 }
-            } catch (Exception e){
-                // As there is no public UI to configure this property,
-                // the value is set to the false to
-                // disable this feature by default
-                // formData.getBoolean throws an exception
-                // if the key to search does not exist
-                // NOTE: Change this when APM Traces was released as public feature.
-                this.setTraceServiceName(DEFAULT_TRACES_SERVICE_NAME);
+                this.setEnableCiVisibility(ciVisibilityData != null && !ciVisibilityData.isNullObject());
+            } catch (Exception ex) {
```

Comment:
would this log a severe error when you raise `throw new FormException("CI Visibility can only be enabled using Datadog Agent mode.", "collectBuildTraces")`? that seems like a mistake

Reply:
If we cannot parse the configuration, it's an error and we should log it as an error IMHO. If we cannot parse the configuration is not an expected behaviour

Reply:
yes but you're also throwing a validation exception in this block and that's not an error in parsing

Reply:
Oh, you're totally right. Good catch.